src/util/metrics.ts

```diff
@@ -26,32 +78,180 @@ export default class Metrics_App {
 	constructor({ host }: IAppConfiguration) {
 		const args = parseArgs();
 
+		this.includeQueryParams = Boolean(args.prometheus_queryparams);
 		this.port = Number(args.prometheus_port);
 		this.app = express();
 		this.host = host;
-
-		this.metricsEndpoint();
+		this.registry = new client.Registry();
+		this.metrics = {};
+		this.init();
 	}
 
 	listen(): void {
 		const { logger } = Log;
 		this.app.listen(this.port, this.host, () => {
-			logger.info(`Metrics Server started at http://${this.host}:${this.port}/`);
+			logger.info(`Metrics Server started at http://${this.host}:${this.port}/metrics`);
 		});
 	}
 
-	/**
-	 * Mount the metrics endpoint.
-	 */
-	private metricsEndpoint() {
-		const register = new client.Registry();
-		register.registerMetric(httpErrorCounter);
-		client.collectDefaultMetrics({ register, prefix: 'sas_' });
+	private createMetricByType(prefix = 'sas', metric: IMetric) {
+		const prefixedName = prefix + '_' + metric.name;
+		if (prefixedName in this.metrics) {
+			return this.metrics[prefixedName];
+		}
+
+		switch (metric.type) {
+			case MetricType.Counter: {
+				const counter = new client.Counter({
+					name: prefixedName,
+					help: metric.help,
+					labelNames: metric.labels || [],
+					registers: [this.registry],
+				});
+
+				this.registry.registerMetric(counter);
+				this.metrics[prefixedName] = counter;
+				return counter;
+			}
+			case MetricType.Histogram: {
+				const histogram = new client.Histogram({
+					name: prefixedName,
+					help: metric.help,
+					labelNames: metric.labels || [],
+					registers: [this.registry],
+					buckets: metric.buckets || [0.1, 0.5, 1, 1.5, 2, 3, 4, 5],
+				});
+
+				this.metrics[prefixedName] = histogram;
+				return histogram;
+			}
+			case MetricType.Gauge:
+				throw new Error('Gauge not implemented');
+			case MetricType.Summary:
+				throw new Error('Summary not implemented');
+			default:
+				throw new Error('Unknown metric type');
+		}
+	}
+
+	private getRoute(req: Request) {
+		let route = req.baseUrl;
+		if (req.route) {
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
```

Comment:
I think we can get the eslint disables removed by explicitly setting the type on `Request` since it takes in a set of Generics: ie `Request<P = core.ParamsDictionary, ResBody = any, ReqBody = any, ReqQuery = qs.ParsedQs, Locals extends Record<string, any> = Record<string, any>>`